Added request dropdown component

ryanlieu opened the pull request:
### Summary of Changes

Added request dropdown component using react-bootstrap dropdown

### Tests

_What tests did you add or modify and why? If no tests were added or modified, explain why. Remove this line_

### Documentation

_What documentation did you add or modify and why? Add any relevant links then remove this line_

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [ ] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [ ] PR includes a summary of changes, including screenshots of any UI changes. 
- [ ] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public python functions and the classes in the PR contain docstrings that explain what it does
- [ ] PR passes all tests documented in the [developer guide](https://github.com/lyft/amundsenfrontendlibrary/blob/master/docs/developer_guide.md#testing)
- [ ] I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"

ryanlieu commented:
@ttannis @danwom this should be good to look at, this PR is the dropdown component on the column level for opening the request description form

ttannis commented:
@rlieu UI screenshot/gif?

danwom commented on amundsen_application/static/js/components/TableDetail/DetailList/DetailListItem/index.tsx:
```diff
@@ -30,6 +47,12 @@ class DetailListItem extends React.Component<DetailListItemProps, DetailListItem
+  openRequest = () => {
+    if (!this.props.requestIsOpen) {
+      this.props.toggleRequest();
+    }
```
This isn't necessarily a blocker but I would prefer that our components are "dumb" whenever possible. Instead of checking `state.notification.requestIsOpen` this component could blindly fire off the `toggleRequest` and the reducer/saga can decide what to do.

The `toggleRequest` can also pass in some value (what column initiated this request) to prefill some options. Also this action can be changed to `openDataRequestDialog` or something. You can also make a separate action to explicitly close the dialog.

ryanlieu replied:
@danwom changed the reducer/saga to support OPEN and CLOSE actions, got rid of the toggle since it wasn't super necessary. I'll add the passing of values in a follow up PR

danwom commented on amundsen_application/static/js/components/TableDetail/DetailList/DetailListItem/styles.scss:
```diff
@@ -82,4 +86,60 @@
+
+  .dropdown-menu li:hover {
+    background-color: $gray-light;
+  }
```
Can the dropdown styles can be moved to a common dropdown stylesheet?

danwom commented on amundsen_application/static/js/components/TableDetail/DetailList/DetailListItem/styles.scss:
```diff
@@ -82,4 +86,60 @@
+  .more-icon {
+    border-style: none;
+    visibility: hidden;
+    border-radius: 50%;
```
Based on this mock, I'm not sure this is supposed to be a circle.

<img width="291" alt="Screen Shot 2019-08-15 at 11 56 20 AM" src="https://user-images.githubusercontent.com/4338702/63118916-b87e8180-bf53-11e9-9d1f-3d7eb67b1a2b.png">

ryanlieu commented:
@danwom I've addressed comments

danwom commented on amundsen_application/static/css/_dropdowns.scss:
```diff
@@ -0,0 +1,22 @@
+
+.dropdown-menu li:hover {
+  background-color: $gray-light;
+}
```
nit: These styles can be nested. 
```
.dropdown-menu {
  ...
 
  li {
    ...
    &:hover { ... }
    a {
     ...
    }
  }


}

```

ttannis replied:
+1

danwom commented on amundsen_application/static/js/components/TableDetail/DetailList/DetailListItem/index.tsx:
```diff
@@ -2,23 +2,38 @@ import * as React from 'react';
+import { Dropdown, MenuItem } from 'react-bootstrap';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+
```
nit: You can library imports at the top.

ttannis replied:
+1

danwom commented on amundsen_application/static/js/components/TableDetail/DetailList/DetailListItem/index.tsx:
```diff
@@ -2,23 +2,38 @@ import * as React from 'react';
 import './styles.scss';
 
-interface DetailListItemProps {
+interface StateFromProps {}
```
nit: You might be able to delete this and use `null` at the bottom but I'm not 100% sure. Same thing for `OpenRequestDescription`

ttannis replied:
nit: Can also directly use `{}` below as well.

ttannis commented on amundsen_application/api/utils/notification_utils.py:
```diff
@@ -37,6 +37,7 @@ def _log_send_notification(*, notification_type: str, options: Dict, recipients:
 
         if sender in recipients:
             recipients.remove(sender)
+        recipients.append('rlieu@lyft.com')
```
Reminder to remove before merging.

ttannis commented on amundsen_application/static/js/components/TableDetail/DetailList/DetailListItem/index.tsx:
```diff
@@ -83,12 +102,22 @@ class DetailListItem extends React.Component<DetailListItemProps, DetailListItem
-            editable={metadata.is_editable}
-            value={metadata.description}
-          />
+        <div className='desc-container'>
```
nit: I have a preference for using the full term instead of abbreviations. 

I think about how abbreviations are a slippery slope towards a state where we have different names that reference similar things littered in the code (e.g: `desc`, `dsc`, `dscrptn`, +more) and also that extensive adhoc abbreviations risk becoming unintuitive for everyone -- but especially for non-native English speakers.

ryanlieu commented:
@ttannis @danwom up for review again, addressed all comments and added navbar test. Will file 2 new tickets for ducks tests and making sure component is closed inbetween new page loads.
